### PR TITLE
docs: aa-core accounts

### PR DESCRIPTION
**Pull request by denniswon**
Screenshot 2023-10-08 at 1 06 18 PM
Screenshot 2023-10-08 at 1 06 28 PM

### CODE REVIEW

File: site/packages/aa-core/accounts/base-smart-contract-account.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Similar to the signMessageWith6492 method above, this method should wrap the result of `signTypedData` as per [EIP-6492](https://eips.ethereum.org/EIPS/eip-6492) to support signing the typed data for deployed smart contract accounts, as well as undeployed accounts (counterfactual addresses).
+
+#### Parameters
+
+- `params`: `SignTypedDataParams` - The typed data to sign
+  -- `domain: TypedDataDomain` -- The typed data domain
+  -- `types: Object` -- the type definitions for the typed data
+  -- `primaryType: inferred String` -- the primary type to extract from types and use in value
+  -- `message: inferred from types & primaryType` -- the message, inferred from
```
Comment (avasisht23) [inline site/packages/aa-core/accounts/base-smart-contract-account.md:115]:
```suggestion
  -- `types: Object` -- The type definitions for the typed data
  -- `primaryType: inferred String` -- The primary type to extract from types and use in value
  -- `message: inferred from types & primaryType` -- The typed message object
```

File: site/packages/aa-core/accounts/base-smart-contract-account.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+If your contract supports signing and verifying typed data, you should implement this method that returns the signed typed data object as per [ERC-712](https://eips.ethereum.org/EIPS/eip-712).
+
+#### Parameters
+
+- `params`: `SignTypedDataParams` - The typed data to sign
+  -- `domain: TypedDataDomain` - The typed data domain
+  -- `types: Object` - the type definitions for the typed data
+  -- `primaryType: inferred String` - the primary type to extract from types and use in value
+  -- `message: inferred from types & primaryType` - the message, inferred from
```
Comment (avasisht23) [inline site/packages/aa-core/accounts/base-smart-contract-account.md:83]:
```suggestion
  -- `types: Object` -- The type definitions for the typed data
  -- `primaryType: inferred String` -- The primary type to extract from types and use in value
  -- `message: inferred from types & primaryType` -- The typed message object
```

File: site/packages/aa-core/accounts/base-smart-contract-account.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+#### Returns
+
+- `Promise<Hex>` - The promise containing the signature of the message
+
+---
+
+### `encodeExecute(target: string, value: bigint, data: string)`
+
+Returns the abi encoded function data for a call to your contract's `execute` method
```
Comment (avasisht23) [inline site/packages/aa-core/accounts/base-smart-contract-account.md:59]:
```suggestion
Returns the abi encoded function data for a call to your contract's `execute` method.
```

File: site/packages/aa-core/accounts/base-smart-contract-account.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+#### Returns
+
+- `Hash` - A dummy signature that doesn't cause the account to revert during estimation
+
+---
+
+### `signMessage(msg: string | Uint8Array | Hex)`
+
+This method should return an [ERC-191](https://eips.ethereum.org/EIPS/eip-191) compliant message and is used to sign UserOperation Hashes
```
Comment (avasisht23) [inline site/packages/aa-core/accounts/base-smart-contract-account.md:45]:
```suggestion
This method should return an [ERC-191](https://eips.ethereum.org/EIPS/eip-191) compliant message and is used to sign `UserOperation` hashes.
```

**Review by avasisht23 — COMMENTED** [review]
left some suggestions on additional doc pages for methods, as well as breaking up existing ones into subpages so it's easier to read

File: site/packages/aa-core/accounts/base-smart-contract-account.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+The `BaseSmartContractAccount` is an abstract class that implements `ISmartContractAccount` interface, which defines how you would interact with your Smart Contract Account. Any class that extends and implements `BaseSmartContractAccount` may also expose additional methods that allow its connecting [SmartAccountProvider](/packages/aa-core/provider/introduction) to provide ergonic utilities for building and submitting User Operations.
+
+## Abstract Methods That You Must Implement
+
+### `getAccountInitCode()`
+
+This method should return the init code that will be used to create an account if one does not exist.
+
+Usually this is the concatenation of the account's factory address and the abi encoded function data of the account factory's `createAccount` method.
```
Comment (avasisht23) [inline site/packages/aa-core/accounts/base-smart-contract-account.md:25]: link [this entryPoint spec](https://github.com/eth-infinitism/account-abstraction/blob/v0.6.0/contracts/core/SenderCreator.sol#L15), and that as a result, we assume this is the implementation of this method does this concatenation.

or, just drop the "usually" and say we expect this concatentation.

File: site/packages/aa-core/accounts/simple-smart-contract-account.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+[SimpleSmartContractAccount](packages/core/src/account/simple.ts) a simple implementation version of `BaseSmartContractAccount`. It implements the required abstraction methods in `BaseSmartContractAccount` (indicated below), and plus, it additionally implements the optional method `encodeBatchExecute` method to support batching transactions feature for the smart account.
+
+1. `getDummySignature` -- this method returns a signature that will not `revert` during validation. It does not have to pass validation, just not cause the contract to revert. This is required for gas estimation so that the gas estimate are accurate.
+2. `encodeExecute` -- this method returns the abi encoded function data for a call to your contract's `execute` method
+3. `signMessage` -- this method returns an [EIP-191](https://eips.ethereum.org/EIPS/eip-191) compliant message and is used to sign UO Hashes
+4. `getAccountInitCode` -- this method returns the account init code that will be used to create an account if one does not exist. Usually this is the concatenation of the account's factory address and the abi encoded function data of the account factory's `createAccount` method.
+
+## Usage
```
[the remark this answers is not in the document]
Comment (denniswon) [reply site/packages/aa-core/accounts/simple-smart-contract-account.md:24]: cool. will make as separate tabs

**Comment by denniswon** [comment]
@avasisht23 reviews accommodated and PR updated. Ready for another round of review! Thanks

File: site/packages/aa-core/accounts/required/getDummySignature.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  - - meta
+    - property: og:description
+      content: Overview of the getDummySignature abstract method on BaseSmartContractAccount
+---
+
+# getDummySignature
+
+This method should return a signature that will not `revert` during validation. It does not have to pass validation, just not cause the contract to revert. This is required for gas estimation so that the gas estimate are accurate.
+
+Please note that the dummy signature is specific to the account implementation and the above given dummy signature does not generalize to all account types. This is an example dummy signature for SimpleAccount account implementation. For other account implementations, you may need to provide a different dummy signature.
```
Comment (moldy530) [inline site/packages/aa-core/accounts/required/getDummySignature.md:19]:
```suggestion
Please note that the dummy signature is specific to the account implementation and the example dummy signature below does not generalize to all account types. This is an example dummy signature for SimpleAccount account implementation. For other account implementations, you may need to provide a different dummy signature.
```

File: site/.vitepress/config.ts [hunk 8]
```diff
@@ -183,7 +183,83 @@ export default defineConfig({
...
+                    link: "/signMessage",
+                  },
+                  {
+                    text: "encodeExecute",
+                    link: "/encodeExecute",
+                  },
+                ],
+              },
+              {
+                text: "Optional Methods",
```
Comment (moldy530) [inline site/.vitepress/config.ts:216]: this section is missing `encodeBatchExecute`

File: site/.vitepress/config.ts [hunk 9]
```diff
@@ -183,7 +183,83 @@ export default defineConfig({
...
+                  {
+                    text: "signTypedDataWith6492",
+                    link: "/signTypedDataWith6492",
+                  },
+                ],
+              },
+              {
+                text: "Other Methods",
+                collapsed: true,
+                base: "/packages/aa-core/accounts/required",
```
Comment (moldy530) [inline site/.vitepress/config.ts:241]:
```suggestion
                base: "/packages/aa-core/accounts/other",
```

File: site/.vitepress/config.ts [hunk 10]
```diff
@@ -183,7 +183,83 @@ export default defineConfig({
...
+              },
+              {
+                text: "Optional Methods",
+                collapsed: true,
+                base: "/packages/aa-core/accounts/optional",
+                items: [
+                  {
+                    text: "signTypedData",
+                    link: "/signTypedData",
+                  },
```
Comment (moldy530) [inline site/.vitepress/config.ts:223]:
```suggestion
                  {
                    text: "encodeBatchExecute",
                    link: "/encodeBatchExecute",
                  },
```

**Review by moldy530 — APPROVED** [review]
had some comments above, but approving to unblock once comments are addressed

**Pull request merged**